eZ/Publish/Core/Repository/SiteAccessAware/Config/IOConfigResolver.php

```diff
@@ -0,0 +1,40 @@
+<?php
+
+/**
+ * @copyright Copyright (C) eZ Systems AS. All rights reserved.
+ * @license For full copyright and license information view LICENSE file distributed with this source code.
+ */
+declare(strict_types=1);
+
+namespace eZ\Publish\Core\Repository\SiteAccessAware\Config;
+
+use eZ\Publish\Core\IO\IOConfig;
+
+/**
+ * @internal
+ */
+final class IOConfigResolver implements IOConfig
+{
+    /** @var string */
+    private $storageDir;
+
+    public function __construct(string $storageDir)
```

Comment:
IMHO `$rootDir`, `$legacyUrlPrefix` and `$urlPrefix` should be separate parameters, even if on DIC level the same value is injected.  

```suggestion
    public function __construct(string $rootDir, string $legacyUrlPrefix, string $urlPrefix)
```